### PR TITLE
Add test for memrchr() & memchr()  #1850

### LINKED ISSUE #1850: Add test for memrchr() & memchr()
Implement some test cases for standard routines memrchr() & memchr()

### PR DESCRIPTION
**Pull request by nakul-shahdadpuri**
Hello,
I have added tests for memchr() and memrchr(). It closes #1850
Please let me know about any changes I should make in my code.
Thank you.

### CODE REVIEW

File: src/tests/stdlib/memchr_test.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <embox/test.h>
+
+EMBOX_TEST_SUITE("test suite for memchr()");
+
+TEST_CASE("Find character in string with memchr()"){
+
```
Comment (anton-bondarev) [inline src/tests/stdlib/memchr_test.c:15]: Rm this line please

File: src/tests/stdlib/memchr_test.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	char str = "ABCDEFGH"
+	char *ps = memrchr(str,'F',strlen(str)); //test case
```
Comment (anton-bondarev) [inline src/tests/stdlib/memchr_test.c:17]: Use the only /* */ as comments
Please fix them for the whole PR

File: src/tests/stdlib/memchr_test.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+	test_assert_not_null(ps);	//checking if a character is found
+	test_assert_zero(strcmp(ps,"FGH"))		//checking if the right character is found
+}
+
+TEST_CASE("Find numeric in numeric string with memchr()"){
+
```
Comment (anton-bondarev) [inline src/tests/stdlib/memchr_test.c:24]: Remove, please

File: src/tests/stdlib/memchr_test.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+TEST_CASE("Find character in zero length string with memchr()"){
+
+	char str = ""
+	char *ps = memrchr(str,'a',strlen(str)); //test case
+
+	test_assert_null(ps);	//checking if a character is not found
+}
+
+
```
Comment (anton-bondarev) [inline src/tests/stdlib/memchr_test.c:48]: Remove trailing empty lines, Please

File: src/tests/stdlib/memchr_test.c [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	char str = "12345678"
+	char *ps = memrchr(str,'4',strlen(str)); //test case
+
+	test_assert_not_null(ps);	//checking if a character is found
+	test_assert_zero(strcmp(ps,"45678"))		//checking if the right character is found
+}
+
+TEST_CASE("Find character in string with no such character with memchr()"){
+
```
Comment (anton-bondarev) [inline src/tests/stdlib/memchr_test.c:33]: Rm please

File: src/tests/stdlib/memrchr_test.c [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+	test_assert_null(ps);	//checking if a character is not found
+}
+
+TEST_CASE("Find character in zero length string with memrchr()"){
+
+	char str = ""
+	char *ps = memrchr(str,'a',strlen(str)); //test case
+
+	test_assert_null(ps);	//checking if a character is not found
+}
```
Comment (anton-bondarev) [inline src/tests/stdlib/memrchr_test.c:46]: Add a new line

File: src/tests/stdlib/memrchr_test.c [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+	char str = "12345678"
+	char *ps = memrchr(str,'4',strlen(str)); //test case
+
+	test_assert_not_null(ps);	//checking if a character is found
+	test_assert_zero(strcmp(ps,"45678"))		//checking if the right character is found
+}
+
+TEST_CASE("Find character in string with no such character with memrchr()"){
+
```
Comment (anton-bondarev) [inline src/tests/stdlib/memrchr_test.c:33]: Remove, please

File: src/tests/stdlib/memrchr_test.c [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <embox/test.h>
+
+EMBOX_TEST_SUITE("test suite for memrchr()");
+
+TEST_CASE("Find character in string with memrchr()"){
+
```
Comment (anton-bondarev) [inline src/tests/stdlib/memrchr_test.c:15]: Remove, please

**Comment by nakul-shahdadpuri** [comment]
Hello, Anton Bondarev
I have made the requested changes.
Thank you.

**Comment by anton-bondarev** [comment]
@nakul-shahdadpuri Thanks.
Please add ampty line in .c sources files

**Comment by anton-bondarev** [comment]
@nakul-shahdadpuri Thanks. Please rebase your PR and squash commits. Then add your tests to /x86/test/unit template with a separate commit

**Comment by anton-bondarev** [comment]
@nakul-shahdadpuri 
But you didn't squash commit just removed commits
Looks at your diffs [https://github.com/embox/embox/pull/1952/files](https://github.com/embox/embox/pull/1952/files)

**Comment by anton-bondarev** [comment]
I've closed PR if you want to contribute these changes please open a new PR

**Pull request closed without merging**